frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -498,8 +502,21 @@ export class TileViewPage extends React.Component {
     const { items, filterGroups, getAvailableFilters, groupByTypes } = this.props;
     const { categories } = this.state;
     const availableFilters = getAvailableFilters(defaultFilters, items, filterGroups);
-
     const activeValues = getActiveValuesFromURL(availableFilters, filterGroups, groupByTypes);
+    if (this.props.storeFilterKey && localStorage.getItem(this.props.storeFilterKey)) {
+      try {
+        const lastFilters = JSON.parse(localStorage.getItem(this.props.storeFilterKey));
+        if (lastFilters && activeValues.activeFilters) {
+          _.each(mapFilterRetention[this.props.storeFilterKey], (filterGroup) => {
+            lastFilters[filterGroup] &&
+              (activeValues.activeFilters[filterGroup] = lastFilters[filterGroup]);
+          });
+        }
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Failed while parsing -> ', e);
+      }
+    }
```

Comment:
The filters from local storage should not override those from the URL. If there are any filters in the URL then those should take precedence. 

Eg. I tried to go directly to a shared url `/catalog/ns/bar?kind=%5B%22Template%22%5D` and my URL filters were ignored.

Reply:
updated